### PR TITLE
Watcher implementation

### PR DESCRIPTION
**Pull request by lasradoVinod**
This code watches for additions of new pods and informs lightfoot.

### CODE REVIEW

File: watcher/watcher.go [hunk 1]
```diff
@@ -13,6 +17,124 @@ import (
...
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 0 || resp.StatusCode > 300 {
+		fmt.Println("Request failed with status:", buffer, resp.Status)
+		return false
+	}
+	return true
+}
+
+func (c *ControllerInfo) ContactLightroom() {
```
Comment (nimf) [inline watcher/watcher.go:82]: too much photo editing, I guess... XD

File: watcher/watcher.go [hunk 2]
```diff
@@ -13,6 +17,124 @@ import (
...
+			controllerInfo.nodes[pod.Spec.NodeName] = pod.Status.PodIP
+			break
+		}
+		for k, v := range pod.ObjectMeta.GetLabels() {
+			if k == "lightfoot" && v == "enable" {
+				done = true
+				break
+			}
+		}
+		if !done {
```
Comment (nimf) [inline watcher/watcher.go:117]: nit: this seems counterintuitive to me. Like if we are not done we should proceed, but we stop instead. Maybe change the variable to "proceed" or "tagged"

File: watcher/watcher.go [hunk 3]
```diff
@@ -13,6 +17,124 @@ import (
...
+	timer := time.NewTimer(10 * time.Second)
+
+	for {
+		select {
+		case <-timer.C:
+		case <-c.wake:
+			c.mu.Lock()
+			for k, _ := range c.pending {
+				status := c.SendUpdate(k)
+				if status {
```
Comment (nimf) [inline watcher/watcher.go:92]: could be `if c.SendUpdate(k) {`

File: watcher/watcher.go [hunk 4]
```diff
@@ -13,6 +17,124 @@ import (
...
+func (c *ControllerInfo) ContactLightroom() {
+	timer := time.NewTimer(10 * time.Second)
+
+	for {
+		select {
+		case <-timer.C:
+		case <-c.wake:
+			c.mu.Lock()
+			for k, _ := range c.pending {
+				status := c.SendUpdate(k)
```
Comment (nimf) [inline watcher/watcher.go:91]: do we want to warn if lightfoot ip is not found for the node repeatedly and pod info keeps hanging in pending?

And overall, logs may be useful even for normal operations.

**Review by nimf — APPROVED** [review]

File: watcher/watcher.go (hunk 1 above)
Comment (lasradoVinod) [reply watcher/watcher.go:82]: Done

File: watcher/watcher.go (hunk 3 above)
Comment (lasradoVinod) [reply watcher/watcher.go:92]: Done

File: watcher/watcher.go (hunk 2 above)
Comment (lasradoVinod) [reply watcher/watcher.go:117]: Done

File: watcher/watcher.go (hunk 4 above)
Comment (lasradoVinod) [reply watcher/watcher.go:91]: Added some more logs

**Pull request merged**